### PR TITLE
Feature/random scenario

### PR DESCRIPTION
**Pull request by hakuturu583**
## Types of PR

- [x] New Features
- [ ] Upgrade of existing features
- [ ] Bugfix

## Description

- Add random scenario example.
- Enable spawn ego entity in C++ scenarios.

## How to review this PR.

Please check passing all CI tests.

### CODE REVIEW

**Review by HansRobo — CHANGES_REQUESTED** [review]
Could you add a sammary of chages to to the release note?
(especially about API changes)

File: mock/cpp_mock_scenarios/src/random_scenario/random001.cpp [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      std::string entity_name = "pedestrian" + std::to_string(i);
+      if (api_.entityExists(entity_name)) {
+        api_.despawn(entity_name);
+      }
+    }
+  }
+
+  void onUpdate() override
+  {
+    [&]() {
```
Comment (HansRobo) [inline mock/cpp_mock_scenarios/src/random_scenario/random001.cpp:113]: Why to use a lambda function here?
I think there is no advantage to use lambda function here than {} to limit scope.

File: mock/cpp_mock_scenarios/src/random_scenario/random001.cpp [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      }
+      if (api_.entityExists(entity_name) && api_.getStandStillDuration(entity_name) >= 0.5) {
+        api_.despawn(entity_name);
+      }
+    };
+    for (int i = 0; i < params_.random_parameters.crossing_pedestrian.number_of_pedestrian; i++) {
+      spawn_and_cross_pedestrian(i);
+    }
+
+    [&]() {
```
Comment (HansRobo) [inline mock/cpp_mock_scenarios/src/random_scenario/random001.cpp:190]: Why to use a lambda function here?
I think there is no advantage to use lambda function here than {} to limit scope.

File: mock/cpp_mock_scenarios/src/random_scenario/random001.cpp [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+            "ego", geometry_msgs::build<geometry_msgs::msg::Pose>()
+                     .position(geometry_msgs::build<geometry_msgs::msg::Point>().x(10).y(-5).z(0))
+                     .orientation(geometry_msgs::msg::Quaternion())),
+          getVehicleParameters(),
+          traffic_simulator::entity::VehicleEntity::BuiltinBehavior::doNothing());
+      }
+      if (!api_.reachPosition("ego", trigger_position, 20.0) && api_.entityExists(entity_name)) {
+        api_.despawn(entity_name);
+      }
+    }();
```
Comment (HansRobo) [inline mock/cpp_mock_scenarios/src/random_scenario/random001.cpp:207]: Why to use a lambda function here?
I think there is no advantage to use lambda function here than {} to limit scope.

File: mock/cpp_mock_scenarios/launch/mock_test.launch.py [hunk 4]
```diff
@@ -77,13 +77,27 @@ def generate_launch_description():
     scenario_package = LaunchConfiguration("package", default="cpp_mock_scenarios")
     junit_path = LaunchConfiguration("junit_path", default="/tmp/output.xunit.xml")
     launch_rviz = LaunchConfiguration("launch_rviz", default=False)
+    vehicle_model = LaunchConfiguration("vehicle_model", default="jpntaxi")
```
Comment (HansRobo) [inline mock/cpp_mock_scenarios/launch/mock_test.launch.py:80]: jpntaxi is not public vehicle description.
In OSS perspective, it is good for default value to use a public vehicle description like lexus or sample_vehicle.

File: mock/cpp_mock_scenarios/src/random_scenario/random001.cpp [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+      if (param_listener_->is_old(params_)) {
+        /// When the parameter was updated, clear entity before re-spawning entity.
+        despawnRoadParkingVehicles();
+        despawnCrossingPedestrians();
+        param_listener_->refresh_dynamic_parameters();
+        params_ = param_listener_->get_params();
+        /// Re-spawn road parking vehicle.
+        spawnRoadParkingVehicles();
+      }
+    }();
```
[the remark this answers is not in the document]
Comment (hakuturu583) [reply mock/cpp_mock_scenarios/src/random_scenario/random001.cpp:123]: Yes, I used the lambda function in order to limit the scope.

File: mock/cpp_mock_scenarios/launch/mock_test.launch.py (hunk 4 above)
Comment (HansRobo) [reply mock/cpp_mock_scenarios/launch/mock_test.launch.py]: @hakuturu583 If you use `sample_vehicle`, you should use `sample_sensor_kit` as sensor_model.

**Review by HansRobo — APPROVED** [review]
LGTM, waiting for green CIs.

**Pull request merged**